pkg/kubenest/controlplane/endpoint.go

```diff
@@ -53,6 +54,30 @@ func installApiServerExternalEndpointInVirtualCluster(dynamicClient dynamic.Inte
 			return errors.Wrap(err, "failed to convert kubernetes endpoint to typed object")
 		}
 
+		// Check if api-server-external-service endpoint already exists
+		_, err = dynamicClient.Resource(schema.GroupVersionResource{
+			Group:    "",
+			Version:  "v1",
+			Resource: "endpoints",
+		}).Namespace(constants.DefaultNs).Get(context.TODO(), constants.ApiServerExternalService, metav1.GetOptions{})
+		if err == nil {
```

Comment:
why not use `apierrors.IsNotFound(err)`

Reply:
"err == nil" is used to check if any type of error has occurred, while "apierrors.IsNotFound(err)" is specifically used to check for the "resource not found" error. Therefore, the scope of "err == nil" is broader than that of "apierrors.IsNotFound(err)." In this case, if an error occurs, there could be multiple types of errors, not just the "resource not found" error.